lib/instrumental_tools/metric_script_executor.rb

```diff
@@ -0,0 +1,96 @@
+require 'benchmark'
+
+class MetricScriptExecutor
+
+  attr_reader :directory, :previous, :last_ran_at
+
+  def initialize(directory)
+    @directory   = directory
+    @previous    = {}
+  end
+
+  def can_execute_file?(path)
+    stat = File::Stat.new(path)
+    stat.executable? && file_is_owner_only?(stat)
+  end
+
+  def can_execute_in_directory?(directory)
+    stat = File::Stat.new(directory)
+    stat.directory? && file_is_owner_only?(stat)
+  end
+
+  def file_is_owner_only?(file_stat)
+    file_stat.owned? && ((file_stat.mode & 0xFFF) ^ 0O700) == 0
+  end
+
+  def run
+    process_to_output = {}
+    if can_execute_in_directory?(directory)
```

Comment:
oh my god such a long if statement - break this method into some smaller methods?